internal/cmd/login/command.go

```diff
@@ -400,8 +356,8 @@ func (c *command) saveLoginToLocal(cmd *cobra.Command, isCloud bool, url string,
 			utils.ErrPrintln(cmd, "The `--save` flag was ignored since SSO credentials are not stored locally.")
 			return nil
 		}
-		ctxName := c.Config.Config.Context().NetrcMachineName // after login, this must be not nil
-		if err := keychain.WriteKeychainItem(ctxName, url, credentials.Username, credentials.Password); err != nil {
+		ctxName := c.Config.Config.Context().LoginContextName // after login, this must be not nil
```

Comment:
login context is a string, unlike state and credentials which are pointers.

Reply:
leaving checking `if cxt!=nil` outside of a getter function is more readable for the code in my opinion

Reply:
We're not checking if the context is `nil` though. Right now we'll panic if it's `nil`.

Reply:
this save function is certain to have a non-nil context, and the other 2 instance of this function have checking `ctx!=nil` outside of this function.

Reply:
We'd save a few lines of code if we checked for `nil` inside the getter function, and all other config file getters use the same approach!

Reply:
`if ctx!=nil` is not checked only for getting login context, but for other purposes as well (see https://github.com/confluentinc/cli/blob/91bef96552dcda9f62954f668ce41c981fe5cdcb/internal/cmd/logout/command.go#L50). having a getter wouldn't allow us to remove this check outside. But I agree i'll add a check inside too to avoid possible panics.